lib/index.js

```diff
@@ -0,0 +1,126 @@
+const fs = require('fs');
+const path = require('path');
+const {
+  fetchLabels,
+  createLabels,
+  deleteLabels,
+  updateLabels,
+} = require('./api');
+
+async function labelControl() {
+  const publicLabels = await fetchLabels();
+  const data = fs.readFileSync(path.join(process.cwd(), 'label-config.json'), {
+    encoding: 'utf8',
+  });
+
+  const localConfig = JSON.parse(data.toString());
+  const illegalLabels = checkForSameLabelName(localConfig);
+  if (illegalLabels.length > 0) {
+    throw new Error(`
+    The config has labels with the same name:
+    ${illegalLabels}
+    The Label names have to be unique.
+    `);
+  }
+  const newLabels = checkForNewOrOldLabels(localConfig, publicLabels);
+  const labelsToDelete = checkForNewOrOldLabels(publicLabels, localConfig);
+  const labelsToUpdate = checkForModifications(localConfig, publicLabels);
+  if (newLabels.length > 0) {
+    await createLabels(newLabels);
+  }
+  if (labelsToDelete.length > 0) {
+    await deleteLabels(labelsToDelete);
+  }
+  if (labelsToUpdate.length > 0) {
+    await updateLabels(labelsToUpdate);
+  }
+  await writeLabelsConfig();
+  const hasChanges =
+    newLabels.length > 0 ||
+    labelsToDelete.length > 0 ||
+    labelsToUpdate.length > 0;
+  const summary = hasChanges
+    ? {
+        message: 'Labels synchronized!',
+        stats: {
+          created: newLabels.length,
+          deleted: labelsToDelete.length,
+          updated: labelsToUpdate.length,
+        },
+      }
+    : { message: 'Labels already up to date!' };
+  return summary;
+}
+
+async function writeLabelsConfig() {
+  const labels = await fetchLabels();
+  fs.writeFileSync(
+    path.join(process.cwd(), 'label-config.json'),
+    JSON.stringify(labels, null, 2)
+  );
+}
+
+function checkForNewOrOldLabels(newLabels, oldLabels) {
+  const oldLabelIds = oldLabels.map(label => {
```

Comment:
It's better to use `filter` here, since an `id` can be undefined

Reply:
Now there are two functions to make it easier to understand
165c084